tests/test_helpers.py

```diff
@@ -38,3 +60,144 @@ def test_uppercases():
+                "DejaVu Sans",
+                "source-sans-pro",
+                "sans-serif",
+            ],
+        ),
+        (
+            "geometric-humanist",
+            [
+                "Avenir",
+                "Montserrat",
+                "Corbel",
+                "URW Gothic",
+                "source-sans-pro",
+                "sans-serif",
+            ],
+        ),
+        ("classical-humanist", ["Optima", "Candara", "Noto Sans", "source-sans-pro", "sans-serif"]),
+        (
+            "neo-grotesque",
+            [
+                "Inter",
+                "Roboto",
+                "Helvetica Neue",
+                "Arial Nova",
+                "Nimbus Sans",
+                "Arial",
+                "sans-serif",
+            ],
+        ),
+        ("monospace-slab-serif", ["Nimbus Mono PS", "Courier New", "monospace"]),
+        (
+            "monospace-code",
+            [
+                "ui-monospace",
+                "Cascadia Code",
+                "Source Code Pro",
+                "Menlo",
+                "Consolas",
+                "DejaVu Sans Mono",
+                "monospace",
+            ],
+        ),
+        (
+            "industrial",
+            [
+                "Bahnschrift",
+                "DIN Alternate",
+                "Franklin Gothic Medium",
+                "Nimbus Sans Narrow",
+                "sans-serif-condensed",
+                "sans-serif",
+            ],
+        ),
+        (
+            "rounded-sans",
+            [
+                "ui-rounded",
+                "Hiragino Maru Gothic ProN",
+                "Quicksand",
+                "Comfortaa",
+                "Manjari",
+                "Arial Rounded MT",
+                "Arial Rounded MT Bold",
+                "Calibri",
+                "source-sans-pro",
+                "sans-serif",
+            ],
+        ),
+        (
+            "slab-serif",
+            [
+                "Rockwell",
+                "Rockwell Nova",
+                "Roboto Slab",
+                "DejaVu Serif",
+                "Sitka Small",
+                "serif",
+            ],
+        ),
+        (
+            "antique",
+            [
+                "Superclarendon",
+                "Bookman Old Style",
+                "URW Bookman",
+                "URW Bookman L",
+                "Georgia Pro",
+                "Georgia",
+                "serif",
+            ],
+        ),
+        (
+            "didone",
+            [
+                "Didot",
+                "Bodoni MT",
+                "Noto Serif Display",
+                "URW Palladio L",
+                "P052",
+                "Sylfaen",
+                "serif",
+            ],
+        ),
+        (
+            "handwritten",
+            ["Segoe Print", "Bradley Hand", "Chilanka", "TSCu_Comic", "casual", "cursive"],
+        ),
+    ],
+)
+def test_get_font_stack_add_emoji_false(name, font_stack):
```

Comment:
This test seems like it has to recode the data in `_get_font_stack()`, which is a bit cumbersome. WDYT of...

* pulling the data in `_get_font_stack` out of the function. For example, `_helpers.py` could have a top-level variable called `FONT_STACKS` that is a dictionary.
* since add emoji does a single action---extend the font list---how about a single test case, checking that add emoji did that in one situation?

This will have the effect that a couple of lines in `_get_font_stack()` can replace all the if/elif font name checks.

Here's an example of what FONT_STACKS my look like in `_helpers.py`:

```python
FONT_STACKS = {
    "system-ui": ["system-ui", "sans-serif"],
    "transitional": ["Charter", ...],
    ...
}

EMOJI_FONTS =  ["Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"]
```

This way, the test of add emoji can just check that the flag does something like...

```

# this could be wrong, but hopefully the gist of it makes sense :o 
# CASE 1:
assert _get_font_stack("system-ui", add_emoji=False) == FONT_STACK["system-ui"]
```

Reply:
@machow , thanks for your review. I'm not sure if the latest commit meets your expectations. I'm happy to hear your feedback if it doesn't.